vignettes/stranded_model.Rmd

```diff
@@ -0,0 +1,139 @@
+---
+title: "Stranded Model - a simulated dataset for machine learning classification"
+author: "Gary Hutson"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{LOS_model: A simulated hospital length-of-stay dataset}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette details why the `stranded_model` dataset was created, how to load it, and gives examples of use with the caret Machine Learning library.
+
+The dataset contains:
+
+- __stranded.label:__ a character metric to indicate whether the patient is stranded, or not
+- __age:__ Integer - the age of the patient on admission to hospital
+- __care.home.referral:__ Integer - flag to indicate referred from care home
+- __medicallysafe:__ Integer - flag to indicate whether the patient is medically safe e.g. safe to be discharged but hasn't been
+- __hcop:__ Integer - flag to indicate whether the patient is in a Health Care for Older People area
+- __mental_health_care:__ Integer - flag to indicate mental health care provision
+- __period_of_previous_care:__ Integer - flag to indicate previous periods of care
+- __admit_date:__ Date - admit date
+- __frailty_index:__ Character - specifying frailty type, if frail
+
+## First, load the data and inspect it
+
+```{r load, warning=FALSE, message=FALSE}
+library(NHSRdatasets)
+library(dplyr)
+library(ggplot2)
+library(caret)
+library(rsample)
+library(varhandle)
+
+#data("stranded_data")
+glimpse(stranded_data)
+prop.table(table(stranded_data$stranded.label))
+```
+
+This is good, it shows a relatively even split between the not stranded and stranded labels. Please refer to the webinar on [Advanced Modelling](https://www.youtube.com/watch?v=rO40vvKXU-4&t=1360s) to look at how you can deal with classification imbalance using techniques such as SMOTE (Synthetic Minority Oversampling Technique Estimation) and ROSE (Random Oversampling Estimation), to name a few. 
+
+## Feature engineering
+
+The next step will be to decide which features need to be engineered for our machine learning model. We will drop the admit_date and recode the frailty index, and perhaps allocate the age into age bands. 
+
+```{r feature_engineering}
+
+stranded_data <- stranded_data %>% 
+  mutate(stranded.label=factor(stranded.label)) %>% 
+  select(everything(), -c(admit_date))
```

Comment:
this is failing - I think there is a conflict on select with dplyr::select being masked. Need to fully qualify all calls to select throughout this Rmd